fix(VncConsole): persist rfc instance in VncConsole

pcbailey opened the pull request:
In the VncConsole component, the 'Disconnect' button and the 'Ctrl+Alt+Del' option in the 'Send Keys' drop down menu do nothing when clicked. This is because the `rfb` instance wasn't persisting properly and was `undefined` when these elements were being clicked. 

This PR stores the `rfb` instance in a ref to ensure it is persisted until the component is unmounted.

Closes # https://github.com/patternfly/patternfly-react/issues/6007

pcbailey commented:
@KKoukiou @yaacov FYI

patternfly-build commented:
PF4 preview: https://patternfly-react-pr-6012.surge.sh

yaacov reviewed (approved):
Hi, we are using this component in openshift console kubevirt plugin
This fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1964789

KKoukiou commented on packages/react-console/src/components/VncConsole/VncConsole.tsx:
```diff
@@ -88,16 +88,18 @@ export const VncConsole: React.FunctionComponent<VncConsoleProps> = ({
-    rfb.addEventListener('securityfailure', onSecurityFailure);
+    if (rfb.current) {
+      rfb.current?.addEventListener('connect', onConnected);
+      rfb.current?.addEventListener('disconnect', onDisconnected);
```
s/onDisconnected/_onDisconnected
s/onSecurityFailure/_onSecurityFailed


So that we update the status with the internal functions.

pcbailey replied:
Nice catch! Fixed! =)

KKoukiou reviewed (changes requested):
Let's hide the Disconnect button when the status is disconnected.
``` js
+      {status !== DISCONNECTED && (
+        <Button variant={ButtonVariant.secondary} onClick={onDisconnect}>
+          {textDisconnect}
+        </Button>
+      )}

```

And one small other issue, see inline in the review.

Then we can add a connect button in the empty state of the disconnected console (of course in another PR) 

Thanks for fixing this.

KKoukiou commented:
I played little bit with it and I think it would be nice to show the 'Connect' button which we currently never show as well. Wanna send the commit within this PR or shall I send it seperately? 

``` js
diff --git packages/react-console/src/components/VncConsole/VncConsole.tsx packages/react-console/src/components/VncConsole/VncConsole.tsx
index cae2230ba..e5317a86c 100644
--- packages/react-console/src/components/VncConsole/VncConsole.tsx
+++ packages/react-console/src/components/VncConsole/VncConsole.tsx
@@ -1,7 +1,7 @@
 import React from 'react';
 
 import { css } from '@patternfly/react-styles';
-import { EmptyState, EmptyStateBody, EmptyStateIcon, Spinner } from '@patternfly/react-core';
+import { Button, EmptyState, EmptyStateBody, EmptyStateIcon, Spinner } from '@patternfly/react-core';
 
 import { initLogging } from '@novnc/novnc/core/util/logging';
 /** Has bad types. https://github.com/larryprice/novnc-core/issues/5 */
@@ -52,6 +52,8 @@ export interface VncConsoleProps extends React.HTMLProps<HTMLDivElement> {
   /** Handshake failed */
   onSecurityFailure?: (e: any) => void;
 
+  /* Text content rendered inside the EmptyState in the "Connect' button for when console is disconnnected */
+  textConnect?: string;
   /* Text content rendered inside the EmptyState for when console is connecting */
   textConnecting?: string | React.ReactNode;
   /* Text content rendered inside the EmptyState for when console is disconnnected */
@@ -82,6 +84,7 @@ export const VncConsole: React.FunctionComponent<VncConsoleProps> = ({
   onDisconnected = () => {},
   onInitFailed,
   onSecurityFailure,
+  textConnect = 'Connect',
   textConnecting = 'Connecting',
   textDisconnected = 'Disconnected',
   textDisconnect = 'Disconnect',
@@ -102,22 +105,26 @@ export const VncConsole: React.FunctionComponent<VncConsoleProps> = ({
     }
   };
 
+  const connect = () => {
+    const protocol = encrypt ? 'wss' : 'ws';
+    const url = `${protocol}://${host}:${port}/${path}`;
+
+    const options = {
+      repeaterID,
+      shared,
+      credentials
+    };
+    rfb.current = new RFB(novncElem, url, options);
+    addEventListeners();
+    rfb.current.viewOnly = viewOnly;
+    rfb.current.scaleViewport = scaleViewport;
+    rfb.current.resizeSession = resizeSession;
+  };
+
   React.useEffect(() => {
     initLogging(vncLogging);
     try {
-      const protocol = encrypt ? 'wss' : 'ws';
-      const url = `${protocol}://${host}:${port}/${path}`;
-
-      const options = {
-        repeaterID,
-        shared,
-        credentials
-      };
-      rfb.current = new RFB(novncElem, url, options);
-      addEventListeners();
-      rfb.current.viewOnly = viewOnly;
-      rfb.current.scaleViewport = scaleViewport;
-      rfb.current.resizeSession = resizeSession;
+      connect();
     } catch (e) {
       onInitFailed && onInitFailed(e);
       rfb.current = undefined;
@@ -185,6 +192,9 @@ export const VncConsole: React.FunctionComponent<VncConsoleProps> = ({
       emptyState = (
         <EmptyState>
           <EmptyStateBody>{textDisconnected}</EmptyStateBody>
+          <Button variant="primary" onClick={connect}>
+            {textConnect}
+          </Button>
         </EmptyState>
       );
       break;
```

pcbailey commented:
@KKoukiou Thanks for the code to add the 'Connect' button! I have updated the commit with your suggestions and also changed the disconnect text to better match SerialConsole.

@yaacov PTAL again when you get a chance. Thanks!

patternfly-build commented:
Your changes have been released in: 
 - @patternfly/react-console@4.10.16

Thanks for your contribution! :tada:
